### PR TITLE
:boom: removing preview3 folder

### LINKED ISSUE #1418: Move csproj-based topics to the main path and redirect preview3 topics
With the migration of .NET Core projects from project.json to the csproj format, we need to restructure the content to no longer have the two paths (preview 2 and preview3+). So the content on the [preview3](https://github.com/dotnet/docs/tree/master/docs/core/preview3) folder need to be moved to its final location and the preview 3 files will be redirected to that place.

### LINKED ISSUE #1634: Remove warnings and links in between versions of the same topic for .NET Core
Topics that are duplicated between core and preview3 have a warning to take users from one topic to the other. For example:

https://docs.microsoft.com/en-us/dotnet/articles/core/tools/dotnet-new
https://docs.microsoft.com/en-us/dotnet/articles/core/preview3/tools/dotnet-new

We need to remove those warnings and links.

### PR DESCRIPTION
**Pull request by mairaw**
## Summary

Removed content from preview3 folder. Removed warnings and links from the top of the articles. Removed RC4 references.

Fixes #1418 and fixes #1634 

## Suggested Reviewers

@blackdwarf @rpetrusha @BillWagner

### CODE REVIEW

File: docs/core/deploying/index.md [hunk 1]
```diff
@@ -182,121 +176,72 @@ Deploying a self-contained deployment with no third-party dependencies involves
...
-    "osx.10.10-x64": {}
-  }
-}
+The following is the complete `csproj` file for this project.
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
```
Comment (BillWagner) [inline docs/core/deploying/index.md]: Should this be `netcoreapp1.1`?

File: docs/core/deploying/index.md [hunk 2]
```diff
@@ -182,121 +176,72 @@ Deploying a self-contained deployment with no third-party dependencies involves
...
-    "osx.10.10-x64": {}
-  }
-}
+The following is the complete csproj file for this project:
+
+```xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>netcoreapp1.0</TargetFramework>
```
Comment (BillWagner) [inline docs/core/deploying/index.md]: same comment: should this be `netcoreapp1.1`?

File: docs/core/tools/cli-msbuild-architecture.md [hunk 3]
```diff
@@ -59,17 +57,14 @@ From an usage perspective, it doesn't change the way you use the CLI. The CLI st
 * `test`
 * `pack` 
 
-These commands still do what you expect them to do (new up a project, build it, publish it, pack it and so on). Majority of the options are not changed, and are still there, and you can consult either the commands' help screens (using `dotnet <command> --help`) or RC4 documentation on this site to get familiar with any changes. 
+These commands still do what you expect them to do (new up a project, build it, publish it, pack it and so on). Majority of the options are not changed, and are still there, and you can consult either the commands' help screens (using `dotent <command> --help`) or documentation on this site to get familiar with any changes.
```
Comment (BillWagner) [inline docs/core/tools/cli-msbuild-architecture.md]: typo: `dotent` should be `dotnet`

**Review by BillWagner — COMMENTED** [review]
Overall, this looks great.  

I may have found a couple version issues, and there's one typo that I found as well.

File: docs/core/tools/cli-msbuild-architecture.md (hunk 3 above)
Comment (mairaw) [inline docs/core/tools/cli-msbuild-architecture.md]: Fixed.

File: docs/core/deploying/index.md (hunk 1 above)
Comment (mairaw) [inline docs/core/deploying/index.md]: Most likely. Changed to 1.1. Also changed the references to 1.0 throughout the text. This topic should probably be reviewed and code moved to snippets folder. I'll open an issue.

File: docs/core/deploying/index.md (hunk 2 above)
Comment (mairaw) [inline docs/core/deploying/index.md]: same line, no?

**Pull request merged**